File: RS2Seminarski.Model/Requests/UserInsertRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RS2Seminarski.Model.Requests
{
    public class UserInsertRequest
    {
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Email { get; set; }
        public string Mobile { get; set; }
        public string UserName { get; set; }
        public string Password { get; set; }
        public string PasswordConfirmation { get; set; }
        public bool? Status { get; set; }


        public List<int> RoleIdList { get; set; } = new List<int> { };
        public List<int> RoutineIdList { get; set; } = new List<int> { };
    }
}

File: ironvault_mobile/lib/model/user_insert_request.dart
import 'package:json_annotation/json_annotation.dart';
part 'user_insert_request.g.dart';

@JsonSerializable()
class UserInsertRequest {
  String? name;
  String? surname;
  String? email;
  String? mobile;
  String? userName;
  String? password;
  String? passwordConfirmation;
  List<String>? roleIdList;

  UserInsertRequest();

  factory UserInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$UserInsertRequestFromJson(json);

  /// Connect the generated [_$UserInsertRequestToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$UserInsertRequestToJson(this);
}
